homeassistant/components/atag/__init__.py

```diff
@@ -0,0 +1,251 @@
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
+            ATTR_ICON: ICONS[TEMP_CELSIUS],
+        },
+        {
+            ATTR_NAME: "Average Outside Temperature",
+            ATTR_ID: "tout_avg",
+            ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS,
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
+            ATTR_ICON: ICONS[TEMP_CELSIUS],
+        },
+        {
+            ATTR_NAME: "Weather Status",
+            ATTR_ID: "weather_status",
+            ATTR_UNIT_OF_MEASUREMENT: None,
+            ATTR_DEVICE_CLASS: None,
+            ATTR_ICON: None,
+        },
+        {
+            ATTR_NAME: "Operation Mode",
+            ATTR_ID: "ch_mode",
+            ATTR_UNIT_OF_MEASUREMENT: None,
+            ATTR_DEVICE_CLASS: None,
+            ATTR_ICON: ICONS[ATTR_MODE],
+        },
+        {
+            ATTR_NAME: "CH Water Pressure",
+            ATTR_ID: "ch_water_pres",
+            ATTR_UNIT_OF_MEASUREMENT: PRESSURE_BAR,
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_PRESSURE,
+            ATTR_ICON: ICONS[PRESSURE_BAR],
+        },
+        {
+            ATTR_NAME: "CH Water Temperature",
+            ATTR_ID: "ch_water_temp",
+            ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS,
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
+            ATTR_ICON: ICONS[TEMP_CELSIUS],
+        },
+        {
+            ATTR_NAME: "CH Return Temperature",
+            ATTR_ID: "ch_return_temp",
+            ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS,
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
+            ATTR_ICON: ICONS[TEMP_CELSIUS],
+        },
+        {
+            ATTR_NAME: "Burning Hours",
+            ATTR_ID: "burning_hours",
+            ATTR_UNIT_OF_MEASUREMENT: HOUR,
+            ATTR_DEVICE_CLASS: None,
+            ATTR_ICON: ICONS[FIRE],
+        },
+        {
+            ATTR_NAME: "Flame",
+            ATTR_ID: "rel_mod_level",
+            ATTR_UNIT_OF_MEASUREMENT: PERCENTAGE,
+            ATTR_DEVICE_CLASS: None,
+            ATTR_ICON: ICONS[FIRE],
+        },
+    ],
+    CLIMATE: {ATTR_NAME: CLIMATE.title()},
+    WATER_HEATER: {ATTR_NAME: "Domestic Hot Water"},
+}
+
+
+async def async_setup(hass: HomeAssistant, config):
+    """Set up the Atag component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Atag integration from a config entry."""
+    session = async_get_clientsession(hass)
+    atag = AtagDataStore(session, paired=True, **entry.data)
+
+    try:
+        await atag.async_update()
+        if not atag.sensordata:
+            raise ConfigEntryNotReady
+    except AtagException:
+        raise ConfigEntryNotReady
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = atag
+
+    for platform in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, platform)
+        )
+
+    async def refresh(event_time):
```

Comment:
Looks like this has a single endpoint.

Can you use `DataUpdateCoordinator` here?
https://developers.home-assistant.io/docs/integration_fetching_data/#coordinated-single-api-poll-for-data-for-all-entities

Reply:
rewrote most of the updaters, hope this is in line? Is it ok to still access the library like I did in Climate and DHW entities?

Reply:
should be fine